src/http/peer.cpp

```diff
@@ -0,0 +1,145 @@
+// Copyright 2022 The Forgotten Server Authors. All rights reserved.
+// Use of this source code is governed by the GPL-2.0 License that can be found in the LICENSE file.
+
+#include "peer.h"
+#include "server.h"
+#include "router.h"
+#include "responder.h"
+#include "../tasks.h"
+
+namespace Http
+{
+
+using Minutes = boost::posix_time::minutes;
```

Comment:
`std::chrono::minutes`

Reply:
`basic_deadline_timer::expires_from_now(const duration_type &expiry_time)` doesn't work with `std::chrono::duration`

Reply:
Doesn't it? We have been using it for ages in `connection.cpp` and others`

Reply:
One is `basic_deadline_timer`, the other is `basic_waitable_timer`, so not the same. I can change it but I'm not sure whether it's appropriate here.